File: iosApp/iosApp/offer/validation/ValidatedNewOfferData.swift
//
//  ValidatedNewOfferData.swift
//  iosApp
//
//  Created by jimmyt on 7/22/22.
//  Copyright © 2022 orgName. All rights reserved.
//

import BigInt
import web3swift

/**
 Validated user-submitted data for a new [Offer](https://www.commuto.xyz/docs/technical-reference/core-tec-ref#offer).
 */
struct ValidatedNewOfferData: Equatable {
    /**
     The contract address of the stablecoin for which the new offer will be made.
     */
    let stablecoin: EthereumAddress
    /**
     The `StablecoinInformation` struct for the stablecoin for which the new offer will be made.
     */
    let stablecoinInformation: StablecoinInformation
    /**
     The minimum stablecoin amount for the new offer.
     */
    let minimumAmount: BigUInt
    /**
     The maximum stablecoin amount for the new offer.
     */
    let maximumAmount: BigUInt
    /**
     The security deposit amount for the new offer.
     */
    let securityDepositAmount: BigUInt
    /**
     The current service fee rate, as a percentage times 100.
     */
    let serviceFeeRate: BigUInt
    /**
     The minimum service fee for the new offer.
     */
    let serviceFeeAmountLowerBound: BigUInt
    /**
     The maximum service fee for the new offer.
     */
    let serviceFeeAmountUpperBound: BigUInt
    /**
     The direction for the new offer.
     */
    let direction: OfferDirection
    /**
     The settlement methods for the new offer.
     */
    let settlementMethods: [SettlementMethod]
    
}

File: androidApp/src/main/java/com/commuto/interfacemobile/android/offer/validation/ValidatedNewOfferData.kt
package com.commuto.interfacemobile.android.offer.validation

import com.commuto.interfacemobile.android.offer.OfferDirection
import com.commuto.interfacemobile.android.offer.SettlementMethod
import com.commuto.interfacemobile.android.ui.StablecoinInformation
import java.math.BigInteger

/**
 * Validated user-submitted data for a new [Offer](https://www.commuto.xyz/docs/technical-reference/core-tec-ref#offer).
 *
 * @property stablecoin The contract address of the stablecoin for which the new offer will be made.
 * @property stablecoinInformation The [StablecoinInformation] for the stablecoin for which the new offer will be made.
 * @property minimumAmount The minimum stablecoin amount for the new offer.
 * @property maximumAmount The maximum stablecoin amount for the new offer.
 * @property securityDepositAmount The security deposit amount for the new offer.
 * @property serviceFeeRate The current service fee rate, as a percentage times 100.
 * @property serviceFeeAmountLowerBound The minimum service fee for the new offer.
 * @property serviceFeeAmountUpperBound The maximum service fee for the new offer.
 * @property direction The direction for the new offer.
 * @property settlementMethods The settlement methods for the new offer.
 */
data class ValidatedNewOfferData(
    val stablecoin: String,
    val stablecoinInformation: StablecoinInformation,
    val minimumAmount: BigInteger,
    val maximumAmount: BigInteger,
    val securityDepositAmount: BigInteger,
    val serviceFeeRate: BigInteger,
    val serviceFeeAmountLowerBound: BigInteger,
    val serviceFeeAmountUpperBound: BigInteger,
    val direction: OfferDirection,
    val settlementMethods: List<SettlementMethod>
)